@here/harp-map-theme/scripts/prepareIcons.ts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+import {
+    AtlasOptions,
+    generateSprites,
+    generateSpritesAtlas,
+    ProcessingOptions
+} from "@here/harp-atlas-tools/src";
+import * as os from "os";
+import * as path from "path";
+
+// Allow to use console output, script runs in a shell (node), not in the browser.
+// tslint:disable:no-console
+
+// tslint:disable-next-line: no-var-requires
+const tmp = require("tmp");
+
+function downloadMakiIcons(targetDir: string): Promise<string> {
+    const ghdownload = require("github-download");
+    const makiPath: string = path.join(__dirname, "..");
+    return new Promise((resolve, reject) => {
+        ghdownload("https://github.com/mapbox/maki#master", targetDir)
+            .on("error", (err: any) => {
+                reject(new Error(err));
+            })
+            .on("end", () => {
+                resolve(path.join(makiPath, targetDir));
+            });
+    });
+}
+
+async function generateMakiAtlas(
+    makiDir: string,
+    atlasPath: string,
+    smallIconsConfig: string,
+    bigIconsConfig: string
+): Promise<any> {
+    const cpus: number = os.cpus() ? os.cpus().length : 4;
+    // Create temprorary directory for sprites output inside makiDir
+    const tmpDir = tmp.dirSync({ template: "sprites-XXXXXX", dir: makiDir });
```

Comment:
tmpDir is not removed afterwards

Reply:
It is, by higher level function which creates root folder for processing, see below:
```
const tmpDir = tmp.dirSync({ template: "resources-tmp-XXXXXX", dir: ".", unsafeCleanup: true });
...

 // Manual cleanup
tmpDir.removeCallback();
```
But maybe I could make it more explicit somehow.

Reply:
It's whether ```dir``` option to tmp.dirSync is not used, or some other reason, but these dirs are created in project root.

Reply:
I have just checked it again on my machine (Ubuntu 16.04). Folders created are logged during processing:
```
Input path: 'resources-tmp-eEJSFC/icons/*-11.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-6T5D7h' 
Input path: 'resources-tmp-eEJSFC/icons/*-15.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-6T5D7h' 
Input path: 'resources-tmp-eEJSFC/sprites-6T5D7h/*.png' 
Output path: 'resources/maki_icons_day[.png|.json]' 
Output dir: 'resources' 
Input path: 'resources-tmp-eEJSFC/icons/*-11.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-JgnfL5' 
Input path: 'resources-tmp-eEJSFC/icons/*-15.svg' 
Output dir: 'resources-tmp-eEJSFC/sprites-JgnfL5' 
Input path: 'resources-tmp-eEJSFC/sprites-JgnfL5/*.png' 
```
The base folder resources-tmp-* is removed after script is done. Could you show me your output?

Reply:
I have strong feeling that this "dir" attribute is ignored on your machine (Windows?), for sake of simplicity and flexibility I have changed implementation a little bit so it should work regardless of this attribute handling. 
Please check it out again.

Reply:
Archlinux actually. Seems to be the issue of ```tmp``` package.